bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -1110,12 +1110,19 @@ void BCStateTran::setEraseMetadataFlagImpl() { psd_->setEraseDataStoreFlag(); }
 
 void BCStateTran::setReconfigurationEngineImpl(std::shared_ptr<ClientReconfigurationEngine> cre) { cre_ = cre; }
 
+template <typename MSG>
+void BCStateTran::stateTransferMsgDeleter(const MSG *m) {
+  const char *p_to_delete = (reinterpret_cast<const char *>(m) - sizeof(MessageBase::Header));
+  std::free(const_cast<char *>(p_to_delete));
+}
+
 // this function can be executed in context of another thread.
 void BCStateTran::handleStateTransferMessageImpl(char *msg,
                                                  uint32_t msgLen,
                                                  uint16_t senderId,
                                                  LocalTimePoint incomingEventsQPushTime) {
   if (!running_) {
+    stateTransferMsgDeleter(msg);
```

Comment:
instead of calling free, take ownership (using unique_ptr) of msg in the first line of the function, then you won't need to free it manually.

Reply:
done